presto-hive/src/main/java/com/facebook/presto/hive/InternalHiveSplit.java

```diff
@@ -168,12 +189,13 @@ public boolean isForceLocalScheduling()
     public InternalHiveBlock currentBlock()
     {
         checkState(!isDone(), "All blocks have been consumed");
-        return blocks.get(currentBlockIndex);
+        List<HostAddress> addresses = blockAddresses.isEmpty() ? ImmutableList.of() : blockAddresses.get(currentBlockIndex);
```

Comment:
nit: Not sure if `blockAddresses.isEmpty() ?` is more clear than `blockAddresses == null`. But it's up to you.

Reply:
i think it's the same. but the empty list is easier for computing size because we don't need to do anything special